Tools/SOGoToolRestore.m

```diff
@@ -172,10 +155,19 @@ - (BOOL) fetchUserID: (NSString *) identifier
   BOOL rc;
   SOGoUserManager *lm;
   NSDictionary *infos;
+  SOGoSystemDefaults *sd;
+  NSString *uid;
 
   lm = [SOGoUserManager sharedUserManager];
   infos = [lm contactInfosForUserWithUIDorEmail: identifier];
-  ASSIGN (userID, [infos objectForKey: @"c_uid"]);
+  sd = [SOGoSystemDefaults sharedSystemDefaults];
```

Comment:
This is failing to raise invalid user when we are in multidomain and the user is not valid, for instance *tralara*. I've fixed by checking the `infos` is not nil before doing the `sd` assignment and set to `nil` `uid` subsequently.